tests/integration/test_private_repos.py

```diff
@@ -59,7 +74,6 @@ def test_private_repos(env_package: str, test_env: dict[str, Any], tmp_path: Pat
     client.download_and_extract_archive(completed_response.id, tmp_path)
     source_path = tmp_path / f"download_{str(completed_response.id)}"
     repo = Repo(source_path / "app")
-    assert repo.head.commit.hexsha == env_data["ref"]
```

Comment:
Can we keep this check (test that the repo returned by Cachito has the ref that we just created and pushed)?